### ISSUE TITLE
Quarto books cross references and search bar not rendering correctly in GitHub Pages

### ISSUE THREAD

**Issue by victorrssx** [quarto-dev/quarto-cli#9196]
### Bug description

When publishing a Quarto book on GitHub Pages, I noticed that the cross-references showed labels instead of the numbers section. For example, **Table 5.1** shows as **Table tbl-data-types**. Another problem that appears is that the search bar does not return any text. Upon investigating deeply I also noticed that this behavior can be found in documents rendered from the `docs` directory. An example of this behaviour can be found at https://ieufrjquant.github.io/intror/objetos.html#tipo-forma

I suppose something is a little strange because local rendering worked fine in the RStudio viewer (and the search bar worked in a new window). I apologize in advance if anything actually went unnoticed by me!

### Expected behavior

![image](https://github.com/quarto-dev/quarto-cli/assets/62582857/a6bc75ff-1b2e-43f0-a8ce-42f643935a31)

![image](https://github.com/quarto-dev/quarto-cli/assets/62582857/556faddc-fe9f-4c90-bfe1-cb98c63730ad)

### Actual behavior

![image](https://github.com/quarto-dev/quarto-cli/assets/62582857/b5f0adcd-dd11-47dd-9f7e-4e53acfba837)

![image](https://github.com/quarto-dev/quarto-cli/assets/62582857/9b2fe74d-325e-46ab-8a21-fd5fcf1cd454)

### Your environment

- IDE: RStudio 2023.12.1+402
- OS: Windows 10

### Quarto check output

```` 
Quarto 1.4.551
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.4.551
      Path: C:\Users\Victor\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\Victor\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2023

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.10.7
      Path: C:/Users/Victor/AppData/Local/Programs/Python/Python310/python.exe
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with py -m pip install jupyter

[>] Checking R installation...........OK
      Version: 4.3.2
      Path: C:/PROGRA~1/R/R-43~1.2
      LibPaths:
        - C:/Users/Victor/AppData/Local/R/win-library/4.3
        - C:/Program Files/R/R-4.3.2/library
      knitr: 1.45
      rmarkdown: 2.25

[>] Checking Knitr engine render......OK
````

**Comment by cscheid** [comment]
Can you point us to a git repo or zip file that allows us to reproduce the issue?

**Comment by victorrssx** [comment]
@cscheid Sure! The public repo is available at https://github.com/ieufrjquant/intror

**Comment by cscheid** [comment]
Sorry - we need a _small_ repro case.

**Comment by victorrssx** [comment]
My bad! I tried to reproduce the errors [in this public repo](https://github.com/victorrssx/intror_test) with a small part of the book but... there it worked perfectly! Strangely, I manage to publish it at https://victorrssx.github.io/intror_test/ with crossrefs and search bar working fine.

So I investigated further and realized that the `quarto render` command gives me (in part) the following output: 

```
pandoc
  to: latex
  output-file: index.tex
  standalone: true
  toc: true
  number-sections: true
  top-level-division: chapter
  pdf-engine: xelatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf

metadata
  crossref:
    chapters: true
  documentclass: scrreprt
  papersize: letter
  classoption:
    - DIV=11
    - numbers=noendperiod
  header-includes:
    - '\KOMAoption{captions}{tableheading}'
  block-headings: true
  lang: pt
  editor: visual
  title: R e RStudio para Iniciantes
  subtitle: Material de Apoio para Cursos Quantitativos do <br> Instituto de Economia da Universidade Federal do Rio de Janeiro (IE/UFRJ)
  author: GPEQ/UFRJ
  date: '2024-03-27T13:37:44-03:00'


Rendering PDF
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode

updating tlmgr

updating existing packages
finding package for multirow.sty
ERROR:
compilation failed- no matching packages
LaTeX Error: File `multirow.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name:
! Emergency stop.
<read *>

l.125 \usepackage
                 {wrapfig}^^M

see E:\Projetos em R\15. Consultoria GPEQ UFRJ\1. Materiais\intror\index.log for more information.

E:\Projetos em R\15. Consultoria GPEQ UFRJ\1. Materiais\intror>git add docs
warning: in the working copy of 'docs/index.html', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'docs/intro.html', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'docs/objetos.html', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'docs/funcoes-e-pacotes.html', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'docs/importando.html', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'docs/instalando-r.html', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'docs/instalando-rstudio.html', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'docs/intro-instalacao.html', LF will be replaced by CRLF the next time Git
touches it
warning: in the working copy of 'docs/intro-lingr.html', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'docs/prim-passos.html', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'docs/site_libs/quarto-diagram/mermaid-init.js', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'docs/site_libs/quarto-diagram/mermaid.css', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'docs/site_libs/quarto-diagram/mermaid.min.js', LF will be replaced by CRLF
the next time Git touches it 
```

It seems to me that something I wrote has resulted in these `warning: in the working copy of 'docs/...', LF will be replaced by CRLF the next time Git touches it` which in turn causes the problems.

**Comment by mcanouil** [comment]
If that is really the root cause, I suggest reading: <https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings>.

**Comment by victorrssx** [comment]
Sorry for being late! FYI: I solved this by 'trial and error'. I selected two random chapters from the book and then observed whether the problem happened again by removing specific parts of them. My conclusion is that it had two main causes.

1. In one of these random selections and modifications I noticed that when removing specific HTML content then it worked. So I added `{.content-visible when-format="html"}`.

2. But when rendering the entire book the problem returned again. Then I realized that another font is related with `kableExtra` package. This package uses some Latex packages that wasn't installed. So I reinstalled `tinytex` and then all those missing packages (the list of everything needed for `kableExtra` can be seen with `kableExtra::kableExtra_latex_packages()`). In reality the fundamental part of the error message was

````
Rendering PDF
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode

updating tlmgr

updating existing packages
finding package for multirow.sty
ERROR:
compilation failed- no matching packages
LaTeX Error: File `multirow.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name:
! Emergency stop.
<read *>

l.125 \usepackage
                 {wrapfig}^^M   
```` 

Hope this helps someone with similar problems.
